src/freenet/support/Fields.java

```diff
@@ -357,6 +358,62 @@ public static String secToDateTime(long time) {
 		return dateString;
 	}
 
+	/**
+	 * Accepted format is {integer}{d|h|min|s}
+	 */
+	public static int durationToMillis(String duration) {
```

Comment:
You can skip reading (for saving time) these 2 methods as they were transferred and rewritten in the next commit.